specs/deneb/polynomial-commitments.md

```diff
@@ -345,22 +317,23 @@ def evaluate_polynomial_in_evaluation_form(polynomial: Polynomial,
     """
     width = len(polynomial)
     assert width == FIELD_ELEMENTS_PER_BLOB
-    inverse_width = bls_modular_inverse(BLSFieldElement(width))
+    inverse_width = BLSFieldElement(width).inverse()
 
     roots_of_unity_brp = bit_reversal_permutation(compute_roots_of_unity(FIELD_ELEMENTS_PER_BLOB))
 
     # If we are asked to evaluate within the domain, we already know the answer
     if z in roots_of_unity_brp:
         eval_index = roots_of_unity_brp.index(z)
-        return BLSFieldElement(polynomial[eval_index])
+        return polynomial[eval_index]
 
-    result = 0
+    result = BLSFieldElement(0)
     for i in range(width):
-        a = BLSFieldElement(int(polynomial[i]) * int(roots_of_unity_brp[i]) % BLS_MODULUS)
-        b = BLSFieldElement((int(BLS_MODULUS) + int(z) - int(roots_of_unity_brp[i])) % BLS_MODULUS)
-        result += int(div(a, b) % BLS_MODULUS)
-    result = result * int(BLS_MODULUS + pow(z, width, BLS_MODULUS) - 1) * int(inverse_width)
-    return BLSFieldElement(result % BLS_MODULUS)
+        a = polynomial[i] * roots_of_unity_brp[i]
+        b = z - roots_of_unity_brp[i]
+        result += a / b
+    r = (BLS_MODULUS + pow(int(z), width, BLS_MODULUS) - 1) % BLS_MODULUS
```

Comment:
Should this stay here?

Reply:
Good catch. Replaced with scalar pow function.